File: php/78-subsets.php
<?php
class Solution {
    private $result = [];
    private $count;
    /**
     * @param Integer[] $nums
     * @return Integer[][]
     */
    function subsets($nums) {
        $this->count = count($nums);
        $this->helper($nums, 0, []);
        return $this->result;
    }
    function helper(&$nums, $start, $val) {
        $this->result[] = $val;
        for ($i = $start; $i < $this->count; $i++) {
            array_push($val, $nums[$i]);
            $this->helper($nums, $i+1, $val);
            array_pop($val);
        }
    }

}

File: go/78-subsets.go
package main

func subsets(nums []int) [][]int {
	res := [][]int{}
	helper(nums, 0, []int{}, &res)
	return res
}
func helper(nums []int, start int, el []int, res *[][]int) {
	*res = append(*res, append([]int{}, el...))
	for i := start; i < len(nums); i++ {
		helper(nums, i+1, append(el, nums[i]), res)
	}
}
